lib/customer.rb

```diff
@@ -20,6 +20,18 @@ def invoices
   end
 
   def transactions
-    repository.find_transactions(id)
+    customer_transactions = invoices.map { |invoice| invoice.transactions }.flatten
   end
+
+  def favorite_merchant
+    successful_transactions = transactions.select { |transaction| transaction.result == "success" }
+
+    successful_invoices = successful_transactions.map { |sv| sv.invoice }
+
+    successful_merchants = successful_invoices.map { |invoice| invoice.merchant }
```

Comment:
Line is too long. [81/80]<br>Pass `&:merchant` as an argument to `map` instead of a block.